spec/requests/submissions_spec.rb

```diff
@@ -111,4 +122,103 @@
       end
     end
   end
+
+  describe "PATCH /submissions/:id" do
+    context "when logged in as a challenge manager" do
+      let(:user) { create_user(role: "challenge_manager") }
+      let(:submission) { create(:submission, challenge: challenge, phase: phase) }
+      let(:evaluator) { create(:user, role: 'evaluator') }
+
+      before do
+        ChallengeManager.create(user: user, challenge: challenge)
+      end
+
+      context "updating judging status" do
+        it "updates eligibility status to selected" do
+          patch submission_path(submission), params: {
+            submission: { judging_status: 'selected' }
+          }, as: :json
+
+          expect(response).to have_http_status(:ok)
+          expect(submission.reload.judging_status).to eq('selected')
+        end
+
+        it "updates advancement status to winner when evaluations are complete" do
+          submission.update!(judging_status: 'selected')
+          create(:evaluator_submission_assignment,
+                 submission: submission,
+                 evaluator: evaluator,
+                 status: :assigned,
+                 evaluation: create(:evaluation, completed_at: Time.current))
+
+          patch submission_path(submission), params: {
+            submission: { judging_status: 'winner' }
+          }, as: :json
+
+          expect(response).to have_http_status(:ok)
+          expect(submission.reload.judging_status).to eq('winner')
+        end
+
+        it "prevents advancement when evaluations are not complete" do
+          submission.update!(judging_status: 'selected')
+          create(:evaluator_submission_assignment,
+                 submission: submission,
+                 evaluator: evaluator,
+                 status: :assigned)
+
+          patch submission_path(submission), params: {
+            submission: { judging_status: 'winner' }
+          }, as: :json
+
+          expect(response).to have_http_status(:unprocessable_entity)
+          expect(submission.reload.judging_status).to eq('selected')
+        end
+
+        it "prevents eligibility changes when evaluators are assigned" do
+          create(:evaluator_submission_assignment,
+                 submission: submission,
+                 evaluator: evaluator,
+                 status: :assigned)
+
+          patch submission_path(submission), params: {
+            submission: { judging_status: 'selected' }
+          }, as: :json
+
+          expect(response).to have_http_status(:unprocessable_entity)
+          expect(submission.reload.judging_status).to eq('not_selected')
```

Comment:
this spec is confusing, I thought we block the update to not_selected when there are assignments? but this tests that we block a judging_status update to selected, which I think is an invalid state (it can't be not_selected if there are assignments)

Reply:
also the title in incorrect since we don't prevent all eligibility changes, only the change from select => not_selected if there are evaluator assignments.